simulator/memory/elf/elf_loader.cpp

```diff
@@ -9,10 +9,11 @@
 #include <elfio/elfio.hpp>
 #include <memory/memory.h>
 
-static void load_elf_section( FuncMemory* memory, const ELFIO::section& section, AddrDiff offset)
+static void load_elf_section( FuncMemory& memory, const ELFIO::section& section, AddrDiff offset)
 {
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast) Connecting ELFIO to our guidelines
-    memory->memcpy_host_to_guest( section.get_address() + offset, reinterpret_cast<const Byte*>(section.get_data()), section.get_size());
+    memory.memcpy_host_to_guest( section.get_address() + offset,
```

Comment:
NOLINTNEXTLINE is no longer effective, as `reinterpret_cast` is moved to the next line.
I think we need a separate statement to define a `const Byte*` pointer.